docs/guides/grafana-vmgateway-openid-configuration.md

```diff
@@ -0,0 +1,162 @@
+      "project_id": 0
+    }
+  }
+}
+```
+
+You can find details about all supported options in the
+vmgateway [documentation](https://docs.victoriametrics.com/vmgateway.html#access-control).
+
+### Configuration for Keycloak
+
+1. Log in with admin credentials to your Keycloak instance
+2. Go to `Clients` -> `Create`.
+   Use `OpenID Connect` as `Client Type`.
+   Specify `grafana` as `Client ID`.
+   Click `Next`.
+   <img src="grafana-vmgateway-openid-configuration/create-client-1.png" width="800">
+3. Enable `Client authentication`.
+   Enable `Authorization`.
+   <img src="grafana-vmgateway-openid-configuration/create-client-2.png" width="800">
+   Click `Next`.
+4. Add Grafana URL as `Valid Redirect URIs`. For example, `http://localhost:3000/`.
+   <img src="grafana-vmgateway-openid-configuration/create-client-3.png" width="800">
+   Click `Save`.
+5. Go to `Clients` -> `grafana` -> `Credentials`.
+   <img src="grafana-vmgateway-openid-configuration/client-secret.png" width="800">
+   Copy the value of `Client secret`. It will be used later in Grafana configuration.
+6. Go to `Clients` -> `grafana` -> `Client scopes`.
+   Click at `grafana-dedicated` -> `Add mapper`.
+   <img src="grafana-vmgateway-openid-configuration/create-mapper-1.png" width="800">
+   <img src="grafana-vmgateway-openid-configuration/create-mapper-2.png" width="800">
+   Configure the mapper as follows
+    - `Mapper Type` as `User Attribute`.
+    - `Name` as `vm_access`.
+    - `Token Claim Name` as `vm_access`.
+    - `User Attribute` as `vm_access`.
+    - `Claim JSON Type` as `JSON`.
+      Enable `Add to ID token` and `Add to access token`.
+      <img src="grafana-vmgateway-openid-configuration/create-mapper-3.png" width="800">
+      Click `Save`.
+7. Go to `Users` -> select user to configure claims -> `Attributes`.
+   Specify `vm_access` as `Key`.
+   Specify `{"tenant_id" : {"account_id": 0, "project_id": 0 }}` as `Value`.
+   <img src="grafana-vmgateway-openid-configuration/user-attributes.png" width="800">
+   Click `Save`.
+
+## Configure grafana
+
+It is required to configure Grafana to use OpenID Connect authentication and to forward the JWT token to vmgateway.
+
+Example configuration for Grafana OpenID Connect authentication:
+```ini
+[auth.generic_oauth]
+enabled = true
+allow_sign_up = true
+team_ids =
+allowed_organizations =
+name = keycloak
+client_id = {CLIENT_ID_FROM_IDENTITY_PROVIDER}
+client_secret = {SECRET_FROM_IDENTITY_PROVIDER}
+scopes = openid profile email
+auth_url = http://localhost:3001/realms/{KEYCLOACK_REALM}/protocol/openid-connect/auth
+token_url = http://localhost:3001/realms/{KEYCLOACK_REALM}/protocol/openid-connect/token
+api_url = http://localhost:3001/realms/{KEYCLOACK_REALM}/protocol/openid-connect/userinfo
+```
+
+Start Grafana. You should be able to log in using your identity provider.
+
+Create a datasource in Grafana. For example, Prometheus datasource with the following URL `http://localhost:8431`.
```

Comment:
```suggestion
Create a new Prometheus datasource in Grafana with the following URL `http://localhost:8431`.
```

what is the URL, though? Should we mention where to find it?
Can we use VictoriaMetrics datasource in the same way?

Reply:
Moved this section to the latter section where vmgateway will be already running and added explanation and VictoriaMetrics datasource reference.